public/app/features/preferences/api/user/baseAPI.ts

```diff
@@ -0,0 +1,36 @@
+import { BaseQueryFn, createApi } from '@reduxjs/toolkit/query/react';
+import { lastValueFrom } from 'rxjs';
+
+import { BackendSrvRequest, getBackendSrv } from '@grafana/runtime';
+
+interface RequestOptions extends BackendSrvRequest {
+  manageError?: (err: unknown) => { error: unknown };
+  showErrorAlert?: boolean;
+  body?: BackendSrvRequest['data'];
+}
+
+function createBackendSrvBaseQuery({ baseURL }: { baseURL: string }): BaseQueryFn<RequestOptions> {
```

Comment:
I think we need to extract this somewhere a it's the same for a lot of the newly generated APIs.

Reply:
Yeah... Every RTKQ slice uses this same function, so really it should be common somewhere... Should be an easy change!

Reply:
Yeah, but where should it be?

Reply:
idk, app/store? It doesn't have to be public, so I don't think it matters all that much.

I think it's something that can be tackeled in a seperate PR!